### PR TITLE
Delete API for transformation_mappings

### PR DESCRIPTION
**Pull request by AparnaKarve**
DELETE API for `transformation_mappings`

```
POST /api/transformation_mappings/<id>
{
  "action" : "delete"
}
```

```
POST /api/transformation_mappings
{
  "action" : "delete",
  "resources" : { 
    'id': 1,
    'id': 2
  }
}
```

```
DELETE /api/transformation_mappings/<id>
```

### CODE REVIEW

File: spec/requests/transformation_mappings_spec.rb [hunk 1]
```diff
@@ -131,6 +131,34 @@
         expect(response.parsed_body).to include(expected)
         expect(response).to have_http_status(:bad_request)
       end
+
```
Comment (abellotti) [inline spec/requests/transformation_mappings_spec.rb]: Hi @AparnaKarve can you add a test that checks that deletes without the appropriate role fails with the forbidden error ?  Thanks.

Comment (AparnaKarve) [reply spec/requests/transformation_mappings_spec.rb]: @abellotti I just added a test for that.
Please review. Thanks.

**Comment by abellotti** [comment]
Thanks @AparnaKarve for the API Enhancement.  LGTM!! 👍

**Comment by himdel** [comment]
@miq-bot add_label transformation

@AparnaKarve is this needed for gaprindashvili or is this for a new post-g feature?

**Comment by AparnaKarve** [comment]
@himdel let's go with the `transformation` label for now. Thanks.

**Comment by AparnaKarve** [comment]
@miq-bot add_label gaprindashvili/yes

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit 65f242cc66cca4bf00b45e5e495ebbd87413b2ad
Author: Alberto Bellotti <abellotti@users.noreply.github.com>
Date:   Mon May 21 12:57:46 2018 -0400

    Merge pull request #383 from AparnaKarve/delete_api_for_transformation_mappings
    
    Delete API for transformation_mappings
    (cherry picked from commit 7fd7c274168f34042107c3f0215e65e6f54e7d0a)
```

**Pull request merged**